components/indexComponents/text.tsx

```diff
@@ -0,0 +1,14 @@
+import router from "next/router";
+import styles from "@/styles/pages.module/index.module.scss";
+
+export function Text() {
+  const go = (e: React.MouseEvent) => {
+    e.preventDefault();
+    router.push("/firmy");
+  };
+  return (
+    <div onClick={go} className={styles.subtitle}>
+      <h1 className={styles.firmyButton}>na naszej stronie znajdziesz...</h1>
+    </div>
+  );
+}
```

Comment:
You can simply use `<Link>` here, no need for `onClick` handlers. `<Link>` will generate an anchor (`<a>`) tag, which is better semantically - it can be focused on using keyboard navigation and it's correctly interpreted by screen readers.
```suggestion
import styles from "@/styles/pages.module/index.module.scss";
import Link from "next/link";

export function Text() {
  return (
    <Link className={styles.subtitle} href="/firmy">
      <h1 className={styles.firmyButton}>na naszej stronie znajdziesz...</h1>
    </Link>
  );
}
```